### ISSUE TITLE
TypeError: 'RemoteReadingResponse' object is not subscriptable

### ISSUE THREAD

**Issue by ohadlevy** [GuyKh/iec-custom-component#91]
### System Health details

Version 	core-2024.3.3
Installation Type 	Home Assistant Supervised
Development 	false
Supervisor 	true
Docker 	true
User 	root
Virtual Environment 	false
Python Version 	3.12.2
Operating System Family 	Linux
Operating System Version 	6.1.21-v8+
CPU Architecture 	aarch64
Timezone 	Asia/Jerusalem
Configuration Directory 	/config

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/ludeeus/iec/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

2024-04-01 10:54:03.008 ERROR (MainThread) [custom_components.iec.coordinator] Unexpected error fetching Iec data: 'RemoteReadingResponse' object is not subscriptable
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 318, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/iec/coordinator.py", line 240, in _async_update_data
    if remote_reading[device.device_number]:
       ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'RemoteReadingResponse' object is not subscriptable


### Reproduction steps

1. I've noticed that since midnight the intergration fails to load
2. I've changed my power supplier at midnight today 
3.
...


### Debug logs

```text
2024-04-01 10:54:03.008 ERROR (MainThread) [custom_components.iec.coordinator] Unexpected error fetching Iec data: 'RemoteReadingResponse' object is not subscriptable
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 318, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/iec/coordinator.py", line 240, in _async_update_data
    if remote_reading[device.device_number]:
       ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'RemoteReadingResponse' object is not subscriptable
```


### Diagnostics dump

_No response_

**Comment by GuyKh** [comment]
Should be fixed in `0.0.24`. Please confirm

**Comment by ohadlevy** [comment]
confirmed! thank you
